### PR TITLE
Add AveragingOp

### PR DESCRIPTION
**Pull request by fzimmermann89**
Calculates the avarage over a dimenion or over subets aong a dimension
Useful for qMRI if the images hve been reconstruced using a sliding windows.
Example cmrf (pseudocode):
````
kdata = kdata.split_k1_into_other(split_indices)
mrf_model = SignalAverage(split_indices) @ CardiacFingerprinting()
````

### CODE REVIEW

**Comment by fzimmermann89** [comment]
Not sure if this is the best abstraction - open for suggestions and docstring changes

We could have a reduction op supporting min, max, mean etc - but only mean is linear.

We could use the maximum index for the domain size if domain_sze has not been set (still ofc with a warning) - assuming that you want to use all data somewhere. But the cmrf reco already breaks that assumption.

File: src/mrpro/operators/AveragingOp.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    """Averaging operator.
+
+    This operator averages the input tensor along a specified dimension.
+    The averaging is performed over groups of elements defined by the `idx` parameter.
+    The output tensor will have the same shape as the input tensor, except for the `dim` dimension,
+    which will have a size equal to the number of groups specified in `idx`. For each group,
+    the average of the elements in that group is computed.
+
+    For example, this operator can be used to simulate the effect of a sliding window average
+    in a signal model.
```
Comment (ckolbPTB) [inline src/mrpro/operators/AveragingOp.py]:
```suggestion
    on a signal model.
```

File: src/mrpro/operators/AveragingOp.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        placeholder = (slice(None),) * (self.dim % x.ndim)
+        averaged = torch.stack([x[(*placeholder, i)].mean(self.dim) for i in self.idx], self.dim)
+        return (averaged,)
+
+    def adjoint(self, x: torch.Tensor) -> tuple[torch.Tensor]:
+        """Apply the adjoint of the averaging operator to the input tensor."""
+        if self.domain_size is None:
+            if self._last_domain_size is None:
+                raise ValueError('Domain size is not set. Please set it explicitly.')
```
Comment (ckolbPTB) [inline src/mrpro/operators/AveragingOp.py]:
```suggestion
                raise ValueError('Domain size is not set. Please set it explicitly or run forward first.')
```

File: tests/operators/test_averagingop.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+from tests import RandomGenerator
+from tests.helper import dotproduct_adjointness_test
+
+
+def test_averageingop_adjointness() -> None:
+    """Test the adjointness of the averaging operator."""
+    rng = RandomGenerator(seed=0)
+    u = rng.complex64_tensor(size=(5, 15, 10))
+    v = rng.complex64_tensor(size=(5, 3, 10))
+    idx = [0, 1, 2], slice(0, 6, 2), torch.tensor([-1, -2, -3])
```
Comment (ckolbPTB) [inline tests/operators/test_averagingop.py:16]: could we also add the case for a boolean Tensor? sth like [False, True, False, True, True] which in `test_averageingop_forward` should be u[(1,3,4),].mean(dim=0), right?

**Review by ckolbPTB — APPROVED** [review]
